config/config.go

```diff
@@ -1,20 +1,44 @@
 package config
 
 import (
+	"bytes"
 	"encoding/json"
 	"errors"
 	"fmt"
 	"net/http"
 	"net/url"
+	"reflect"
 	"time"
 )
 
+type Body struct {
+	Type    string                 `json:"type"`
+	Content map[string]interface{} `json:"content"`
+}
+
+func NewBody(bodyType, bodyContent string) (*Body, error) {
+	var body Body
+	body.Type = bodyType
+	bodyContentMap := make(map[string]interface{})
+	if !reflect.DeepEqual("", bodyContent) {
+		err := json.Unmarshal([]byte(bodyContent), &bodyContentMap)
+		if err != nil {
+			return nil, errors.New("bodyContent is not valid json data")
+		}
+	}
+	body.Content = bodyContentMap
+	return &body, nil
+}
+
+var contentTypeValidValues = []string{"application/json"}
```

Comment:
I thought we would treat the input `body` as raw text when we discussed about it 🤔 
It's simpler for us not to know what it is, and I don't think we should perform json validation: what if the user wants to test their endpoint with an invalid json?
Same for the content type: we probably don't want to be smart and dictate it: the user should be able to build its own request without surprises

Reply:
To be honest it was super blurry to me too, even when I wrote the issue (note the number of "may" 😅).

But i agree that just copy pasting the body content provided is a bit easier and gives more freedom to the user (not our fault their requests fail, runs on its machine anyway!).